okhttp/src/main/java/okhttp3/OkHttpClient.java

```diff
@@ -460,6 +463,21 @@ public Builder() {
       connectionSpecs = DEFAULT_CONNECTION_SPECS;
       eventListenerFactory = EventListener.factory(EventListener.NONE);
       proxySelector = ProxySelector.getDefault();
+      if (proxySelector == null) {
+        proxySelector = new ProxySelector() {
```

Comment:
I'd suggest moving this to a named class under the internal package.  No reason to have this logic inside OkHttpClient, it's big enough already.

Reply:
That actually was my first thought. :-)
I was just not sure where exactly and how to name it to fit your conventions, so I made it like this waiting for complaints. :-D
`internal` directly? Or `internal/connection`? Or `internal/proxy`?
What would be a good class name that fits your naming policies?
`NoProxySelector`?

Reply:
I liked NullProxySelector that @swankjesse suggested. Either internal/http or internal/proxy would be fine for me.  No strong opinion from me.

Reply:
Ok, then I think `internal/proxy` is better fit.
Well, I don't like `NullProxySelector` as it does not select a `null` proxy, it selects the `NO_PROXY` proxy that stands for a direct connection.
That it is used if the system default proxy selector is `null` is not really significant imho.
But if you prefer that name, I'll change it to that, it's your project. :-)